src/lib/util.ts

```diff
@@ -50,8 +52,32 @@ export default class Util {
   public static getWeb3() {
     return getArc().web3;
   }
-  public static async getNetworkId() {
-    return await getArc().web3.eth.net.getId();
+
+  /**
+   * return network id, independent of the presence of Arc
+   * @param web3Provider
+   */
+  public static async getNetworkId(web3Provider?: any): Promise<string> {
+    let arc: any;
+    let web3: any;
+
+    try {
+      arc = getArc();
+    } catch {}
+
+    if (arc) {
+      web3 = arc.web3;
+    } else if ((<any> window).web3) {
+      web3 = (<any> window).web3;
+    } else if (web3Provider) {
+      web3 = new Web3(web3Provider);
+    }
```

Comment:
So this code will try to find the web3provider on the window object by metamask or gnosis safe.

The `getMetamask`  function which is used to find the web3provider to pass to Arc does the same thing (it looks at the `window.ethereum` object). It is probably better to use the same function also here, so we are sure that we pass the same web3 provider to Arc as we are using here to check the id.

Reply:
That code isn't actually looking for web3Provider here.  web3provider  is optionally passed-in.  The code is looking for window.web3.

Reply:
what I mean is that what we are interested in here is to get info about the network that we are connected to (which is what the web3 provider does).

The `getMetamask` function, that is used for the same purpose - find the injected web3 instance, get information about the provider, and pass that on to Arc.

The logic of the two methods is slightly different (get the window.ethereum object vs the window.web3 object), and it should be the same.

Reply:
... and the provider that is passed-in is the same one that we'll send to Arc

Reply:
OK I simplified the use of getMetaMask a bit, and added the code from above to firm up getNetworkId